lib/helpers/getPrerenderManifest.js

```diff
@@ -1,9 +1,46 @@
 const { join } = require("path");
 const { readJSONSync } = require("fs-extra");
 const { NEXT_DIST_DIR } = require("../config");
+const nextConfig = require("./getNextConfig")();
+const getDataRouteForRoute = require("./getDataRouteForRoute");
+
+const transformManifestForI18n = (manifest) => {
+  const { routes, dynamicRoutes } = manifest;
+  const { defaultLocale, locales } = nextConfig.i18n;
+  const newRoutes = {};
+  Object.entries(routes).forEach(
+    ([route, { dataRoute, initialRevalidateSeconds, srcRoute }]) => {
+      const isDynamicRoute = !!srcRoute;
+      if (isDynamicRoute) {
+        newRoutes[route] = routes[route];
+      } else {
+        locales.forEach((locale) => {
+          const routeWithPrependedLocale = `/${locale}${route}`;
+          newRoutes[routeWithPrependedLocale] = {
+            dataRoute: getDataRouteForRoute(route, locale),
+            srcRoute: route,
+            initialRevalidateSeconds: false,
```

Comment:
Couldn't you achieve the same result by moving this logic into `getStaticProps/pages.js`? It's the only one that uses the `routes` entry, as far as I know.

Reply:
1. made the params spread change
2. may circle back to this question (re: getPrerenderManifest logic vs getStaticProps/pages logic) after i got through the rest of your review, but at this moment (and i think we briefly discussed this on slack) all the pages using GSP (aka reval + fallback) depend on logic in this transformation right now

Reply:
now that i removed the dynamicRoutes logic from this transformation, it's now down to just plain GSP and GSPwithReval pages depending on this logic. i don't see a major benefit to moving it into a helper and including it in pages right now so i think this will work for now since they both use prerender-manifest's `routes`